crates/agent-sql/src/text_json.rs

```diff
@@ -33,19 +33,17 @@ impl<T> sqlx::postgres::PgHasArrayType for TextJson<T> {
     }
 }
 
-impl<T: Serialize> Encode<'_, postgres::Postgres> for TextJson<T> {
+// Encode and Decode pass through to the sqlx::types::Json implementation,
+// but are restricted to the JSON (and not JSONB) postgres types.
+impl<'q, T: Serialize> Encode<'q, postgres::Postgres> for TextJson<T> {
```

Comment:
This ended up not actually preventing the leading space from being encoded with JSON in the database (sqlx does it anyway), but felt a bit cleaner so I left it in.

Reply:
How did you end up resolving this? I'm not seeing specific handling and I see you added a round-trip test to confirm it's not modified.

Reply:
The column that calculates the hash does a `trim` on the value before hashing it to remove the leading space, see [`supabase/migrations/09_publications.sql`](https://github.com/estuary/flow/pull/926/commits/a7ba22fdd55d66beefa1e27b1824c086e2ad6aee#diff-8e4208014719024769b27f0fba338b404fd0f070e1f8f7cd3716afcde7c3946bR58)